cmd/root.go

```diff
@@ -129,20 +132,33 @@ func getImage(imageName string) (pkgutil.Image, error) {
 	var cachePath string
 	var err error
 	if !noCache {
-		cachePath, err = cacheDir(imageName)
+		cachePath, err = getCacheDir(imageName)
 		if err != nil {
 			return pkgutil.Image{}, err
 		}
 	}
 	return pkgutil.GetImage(imageName, includeLayers(), cachePath)
 }
 
-func cacheDir(imageName string) (string, error) {
-	dir, err := homedir.Dir()
-	if err != nil {
-		return "", err
+func getCacheDir(imageName string) (string, error) {
+
+	// First preference for cache is set at command line
+	if cacheDir == "" {
+
+		// second preference is environment
+		cacheDir = os.Getenv(containerDiffEnvCacheDir)
 	}
-	rootDir := filepath.Join(dir, ".container-diff", "cache")
+
+	// Third preference (default) is set at $HOME
+	if cacheDir == "" {
+		dir, err := homedir.Dir()
+		if err != nil {
+			return "", err
```

Comment:
let's wrap the error here (for more context)

`return "", errors.Wrap(err, "retrieving home dir")`

Reply:
Check on how I did this to make sure it is what you intended, replacing the return statement with a more detailed one.

Reply:
it looks like there is issue with this line, although the error output isn't meaningful (undefined errors). I'll revert back for now and we can have further discussion.

Reply:
not totally sure what the error you were getting was, but maybe it was an import error? what you had before looked right to me.

if you don't have auto-imports on, you'll need to import `github.com/pkg/errors` to use `errors.Wrap()`. could you try that and see if it works?

Reply:
That's totally correct! I took the error message at face value, as in "there are errors and they are undefined" (and I didn't know how to resolve) as opposed to "there is a package called errors and you didn't install it so it's undefined." Here is the "errors error" before:

```bash
$ make test
GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -tags "container_image_ostree_stub containers_image_openpgp" -ldflags "-X github.com/GoogleContainerTools/container-diff/version.version=v0.13.0" -o out/container-diff-linux-amd64 github.com/GoogleContainerTools/container-diff
# github.com/GoogleContainerTools/container-diff/cmd
cmd/root.go:154:15: undefined: errors
Makefile:45: recipe for target 'out/container-diff-linux-amd64' failed
make: *** [out/container-diff-linux-amd64] Error 2
```
and after including the package, works :) The final snippet with the requested change is now:

```go
	if cacheDir == "" {
		dir, err := homedir.Dir()
		if err != nil {
			return "", errors.Wrap(err, "retrieving home dir")
		} else {
			cacheDir = dir
		}
	}
```
I wonder how many others have been confused by that error message because of the naming :thinking: I find this very funny. :)

I will commit and push soon after reviewing the other issues.

Reply:
Random question - what is the difference between:

```go
import (
	"errors"
)
```
and

```go
import (
	"github.com/pkg/errors"
)
```
Are they two references to the same thing? Or does the latter be more specific about the import, ensuring it's from a particular namespace?

Reply:
haha yeah, errors-related error messages are the worst. glad you got it figured out :)

the `errors` package is a built in [library](https://golang.org/pkg/errors/), which basically just allows you to create new errors. not a lot of functionality. 

[`github.com/pkg/errors`](https://godoc.org/github.com/pkg/errors) on the other hand has a bit more versatility, making it easy to wrap error messages with a bit more context, as well as providing stack traces and a bit more contextual information. we usually prefer using it whenever possible, because nobody likes vague error messages :)

Reply:
Cool thanks, this is good to know!